[ME-1088] Update npm packages to resolve Vanta vulnerabilities for SOC2 compliance

aalirehman opened the pull request:
@davidswaven @antoinegoutagny @Waynezhi

To prepare for SOC2 compliance, Vanta has outlined some repositories with outdated packages. I went ahead and tried to resolve any peer dependency vulnerabilities by running `npm audit fix`

https://mikmakai.atlassian.net/browse/ME-1088

`mongoose` is a dev dependency so we can ignore that
`braces` is a peer dependency used by `mocha` which is a dev dependency so we can ignore that

I wasn't able to locate where `ip` was being used, so I'm hoping `npm audit fix` updated it correctly.

Here is the output of npm audit:

```
[15:53:21] ➜  node-db-connector git:(master) ✗ npm audit                         
# npm audit report

braces  <3.0.3
Severity: high
Uncontrolled resource consumption in braces - https://github.com/advisories/GHSA-grv7-fg5c-xmjg
fix available via `npm audit fix`
node_modules/braces

get-func-name  <2.0.1
Severity: high
Chaijs/get-func-name vulnerable to ReDoS - https://github.com/advisories/GHSA-4q6p-r6v2-jvc5
fix available via `npm audit fix`
node_modules/get-func-name

ip  *
Severity: high
ip SSRF improper categorization in isPublic - https://github.com/advisories/GHSA-2p57-rm9w-gvfp
fix available via `npm audit fix`
node_modules/ip
  socks  1.0.0 - 2.7.1
  Depends on vulnerable versions of ip
  node_modules/socks

mongodb  5.0.0 - 5.7.0
Severity: moderate
MongoDB Driver may publish events containing authentication-related data - https://github.com/advisories/GHSA-vxvm-qww3-2fh7
fix available via `npm audit fix`
node_modules/mongoose/node_modules/mongodb
  mongoose  0.0.3 - 0.0.6 || 7.0.0-rc0 - 7.8.2
  Depends on vulnerable versions of mongodb
  node_modules/mongoose


nanoid  <3.3.8
Infinite loop in nanoid - https://github.com/advisories/GHSA-mwcw-c2x4-8c55
fix available via `npm audit fix`
node_modules/nanoid
  mocha  8.2.0 - 10.2.0
  Depends on vulnerable versions of nanoid
  node_modules/mocha

8 vulnerabilities (2 low, 1 moderate, 5 high)

To address all issues, run:
  npm audit fix

```

antoinegoutagny commented:
- `mongoose` is not a direct dependency, a reference must be passed by the code using this package.
- `ip` is a sub-dependency and not used directly
- current version is already 6.1.2, looks like the package-lock is not in sync with package.json. Cf. https://www.npmjs.com/package/node-db-connector.
- package.json still references mongodb v4 not v5, and should be updated.

aalirehman commented:
> * `mongoose` is not a direct dependency, a reference must be passed by the code using this package.
> 
>     * `ip` is a sub-dependency and not used directly
> 
>     * current version is already 6.1.2, looks like the package-lock is not in sync with package.json. Cf. https://www.npmjs.com/package/node-db-connector.
> 
>     * package.json still references mongodb v4 not v5, and should be updated.

@antoinegoutagny 

- Updated the version to 6.1.3 using `npm version patch`
- Updated MongoDB to v5.9.2
